mapie/calibration.py

```diff
@@ -0,0 +1,559 @@
+            as the class labels.
+            Read more in the :ref:`User Guide <stratification>`.
+
+        Returns
+        -------
+        MapieCalibrator
+            The model itself.
+        """
+        self._check_method()
+        cv = check_cv(self.cv)
+        estimator = check_estimator_classification(X, y, cv, self.estimator)
+        calibrator = self._get_calibrator(self.calibrator)
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        assert type_of_target(y) in ["multiclass", "binary"]
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        self.n_features_in_ = check_n_features_in(X, cv, estimator)
+        random_state = check_random_state(random_state)
+
+        if cv == "prefit":
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.n_classes_ = len(self.classes_)
+            self.calibrators = self._fit_calibrators(
+                X, y, sample_weight, calibrator
+            )
+        if cv == "split":
+            results = get_calib_set(
+                X,
+                y,
+                sample_weight=sample_weight,
+                calib_size=calib_size,
+                random_state=random_state,
+                shuffle=shuffle,
+                stratify=stratify,
+            )
+            X_train, y_train, X_calib, y_calib, sw_train, sw_calib = results
+            X_train, y_train = indexable(X_train, y_train)
+            y_train = _check_y(y_train)
+            sw_train, X_train, y_train = check_null_weight(
+                sw_train,
+                X_train,
+                y_train
+            )
+            self.n_classes_ = len(np.unique(y_train))
+            self.estimator = fit_estimator(
+                clone(estimator), X_train, y_train, sw_train,
+            )
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.calibrators = self._fit_calibrators(
+                X_calib, y_calib, sw_calib, calibrator
+            )
+        return self
+
+    def predict_proba(
+        self,
+        X: ArrayLike,
+    ) -> NDArray:
+        """
+        Prediction of the calibrated scores using fitted classifer and
+        calibrator.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes)
+            The calibrated score for each max score and zeros at every
+            other position in that line.
+        """
+        check_is_fitted(self, self.fit_attributes)
+        self.uncalib_pred = self.estimator.predict_proba(X=X)  # type: ignore
+
+        max_prob, y_pred = self._get_labels(X)
+
+        n = _num_samples(max_prob)
+        calibrated_test_values = np.zeros((n, self.n_classes_))
+
+        for idx, item in enumerate(np.unique(y_pred)):
+            calibrated_test_values = self._pred_proba_calib(
```

Comment:
you change the whole array, not only slices

Reply:
True, in this case should not be an issue as there should only be one maximum per line and therefore, no line should ever be re-written over another.